eosdis-nasa/earthdata-pub-api
# Edpub 1173 user metrics v2

# Description

Add the capability the trac advanced submission and step metrics and refresh the generate metrics endpoint to support these metrics in a more organized fashion.

## Spec

See Ticket: [EDPUB-1173](https://bugs.earthdata.nasa.gov/browse/EDPUB-1173), [EDPUB-1212](https://bugs.earthdata.nasa.gov/browse/EDPUB-1212), [EDPUB-1211](https://bugs.earthdata.nasa.gov/browse/EDPUB-1211)

---

## Validation

1. Make sure all merge request checks have passed (CI/CD).
2. Pull related branches locally.
3. Navigate to the local dashboard.
4. Across at least 2 DAACs create 1 submission per Daac that is published, 1 submission that is in progress, 1 with a rejected accession request, and 1 that is withdrawn.
5. Create at least 2 users one that is a data producer and one that is DAAC staff.
6. Using the documentation availible [here](https://docs.google.com/document/d/1Wd0KAoGPgP3mFpi4i8vgdTEAIk9EX0XrwRykxvumK24/edit) confirm all metrics and filters are working as expected.
7. Confirm an admin can see all metrics, DAAC staff can only see there DAAC and a data producer has no access.

---

## Change Log

Add the capability the trac advanced submission and step metrics and refresh the generate metrics endpoint to support these metrics in a more organized fashion.

* [Add _____](link to commitid)
